### PR TITLE
SWWAE Example: Conv kernel size in resid pathway to 1x1 and activation from BN+RELU to ELU

### PR DESCRIPTION
**Pull request by antonmbk**
The original thread: [https://github.com/fchollet/keras/pull/5722](url)

Long story short, cleaned up resid block by making the resid pathway kernel size 1x1 and added ELU activation instead of the slower BN+RELU. The option to do either one is available; obviously each has its own advantages.

In the end (@nouiz), on Windows 10 with Theano 0.9.0rc4 and cudnn 5 I had the following results for each activation type:

ELU:

Using gpu device 0: GeForce GTX 780 (CNMeM is disabled, cuDNN 5005)
x_train shape: (60000L, 1L, 28L, 28L)
60000 train samples
10000 test samples
Train on 60000 samples, validate on 10000 samples
Epoch 1/5
60000/60000 [==============================] - 69s - loss: 0.0247 - val_loss: 0.0062
Epoch 2/5
60000/60000 [==============================] - 71s - loss: 0.0049 - val_loss: 0.0041
Epoch 3/5
60000/60000 [==============================] - 74s - loss: 0.0037 - val_loss: 0.0033
Epoch 4/5
60000/60000 [==============================] - 76s - loss: 0.0030 - val_loss: 0.0027
Epoch 5/5
60000/60000 [==============================] - 76s - loss: 0.0025 - val_loss: 0.0023

BN+RELU:
Using gpu device 0: GeForce GTX 780 (CNMeM is disabled, cuDNN 5005)
x_train shape: (60000L, 1L, 28L, 28L)
60000 train samples
10000 test samples
Train on 60000 samples, validate on 10000 samples
Epoch 1/5
60000/60000 [==============================] - 104s - loss: 0.0364 - val_loss: 0.0094
Epoch 2/5
60000/60000 [==============================] - 106s - loss: 0.0071 - val_loss: 0.0057
Epoch 3/5
60000/60000 [==============================] - 108s - loss: 0.0052 - val_loss: 0.0049
Epoch 4/5
60000/60000 [==============================] - 108s - loss: 0.0043 - val_loss: 0.0040
Epoch 5/5
60000/60000 [==============================] - 111s - loss: 0.0038 - val_loss: 0.0036

### CODE REVIEW

File: examples/mnist_swwae.py [hunk 1]
```diff
@@ -51,20 +51,23 @@
...
 from keras.layers import UpSampling2D, Conv2D, MaxPooling2D
-from keras.layers import Input, BatchNormalization
+from keras.layers import Input, BatchNormalization, ELU
 import matplotlib.pyplot as plt
 import keras.backend as K
 from keras import layers
 
 
-def convresblock(x, nfeats=8, ksize=3, nskipped=2):
+def convresblock(x, nfeats=8, ksize=3, nskipped=2, elu=True):
```
Comment (fchollet) [inline examples/mnist_swwae.py]: Please add a proper docstring detailing the two different behaviors (see other docstrings in Keras for examples of the format).

**Comment by fchollet** [comment]
LGTM, thanks!
